vcs/gitcmd/repo.go

```diff
@@ -161,105 +161,123 @@ func (r *Repository) ResolveTag(name string) (vcs.CommitID, error) {
 	return commitID, nil
 }
 
-// branchCounts returns the behind/ahead commit counts information for branch, against base branch.
-func (r *Repository) branchCounts(branch, base string) (behind, ahead uint, err error) {
-	if err := checkSpecArgSafety(branch); err != nil {
-		return 0, 0, err
-	}
-	if err := checkSpecArgSafety(base); err != nil {
-		return 0, 0, err
-	}
+// stringFilter helps manage a list of string filters.
+type stringFilter struct{ only map[string]struct{} }
```

Comment:
I like the idea of having this `stringFilter` and using it to filter out branches.

I have a suggestion/idea for how to make its implementation slightly shorter and simpler. It assumes that `stringFilter` is "done" and won't need to change/gain extra features in the future, which seems reasonable given that it's a specialized helper for a specific task here.

If so, consider changing its definition from:

```Go
// stringFilter helps manage a list of string filters.
type stringFilter struct{ only map[string]struct{} }
```

To:

```Go
// branchFilter is a filter for branch names.
// If not empty, only contained branch names are allowed. If empty, all name are allowed.
// The map should be made so it's not nil.
type branchFilter map[string]struct{}
```

Then it's more natural to define it as:

```Go
var f = make(branchFilter)
```

The implementation of `allows` method can remain as is.

But the implementation of `add` method can be changed to:

```Go
func (only branchFilter) add(list []string) {
	for _, l := range list {
		only[l] = struct{}{}
	}
}
```

Basically, removing the `if f.only == nil { f.only = make(map[string]struct{}) }` initialization. That just seems out of place for `add` in my opinion. It feels more explicit to make it once when declaring the filter.

However, that is largely a preference and the difference is quite small, so feel free to go with what you think is better overall.